commit 3d9a07845ffca3f5c5ae542d58912f3970542b61
Author: hasufell <hasufell@gentoo.org>
Date:   Thu Jul 18 14:03:48 2013 +0200

    BUILD: reorder flags
    
    This enables environment overrides without the need
    for patching such as "-fno-visibility=hidden".

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index dc563c0..f75d8ab 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -3,9 +3,9 @@ if(ENABLE_WERROR)
 	set(WHISKERMENU_FLAGS "${WHISKERMENU_FLAGS} -Werror")
 endif()
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${WHISKERMENU_FLAGS}")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden -std=c++0x ${WHISKERMENU_FLAGS}")
+set(CMAKE_CXX_FLAGS "-fvisibility=hidden -std=c++0x ${WHISKERMENU_FLAGS} ${CMAKE_CXX_FLAGS}")
 if(ENABLE_AS_NEEDED)
-	set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--as-needed")
+	set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--as-needed ${CMAKE_SHARED_LINKER_FLAGS}")
 endif()
 
 find_package(GTK2 REQUIRED)
